app/src/main/java/com/example/dennis/leagueoflegions/GameView.java

```diff
@@ -85,16 +107,30 @@ private void draw()
             ArrayList<Player> players = game.getPlayers();
             for(Player player : players) {
                 for(Unit unit : player.getUnits()) {
-                    unitPaint.setColor(unit.getColor());
-                    canvas.drawRect(unit.getRect(), unitPaint);
-                    if (!unit.getPath().isEmpty()) {
-                        pathPaint.setColor(unit.getColor());
-                        pathPaint.setAlpha(64);
+                    if(unit.getType().equals("rectangle")) {
+                        unitPaint.setColor(unit.getColor());
+                        canvas.drawRect(unit.getRect(), unitPaint);
+                        if (!unit.getPath().isEmpty()) {
+                            pathPaint.setColor(unit.getColor());
+                            pathPaint.setAlpha(64);
+                            canvas.drawPath(unit.getRemainingPath(), pathPaint);
+                        }
+                    }
+                    else {
+                        int x = unit.getX();
+                        int y = unit.getY();
+
+                        for(int i = -10; i <= 10; i++) {
+                            for (int j = -10; j <= 10; j++) {
+                                canvas.drawCircle(x + 20*i, y + 20*j, 5, unitPaint);
+                            }
+                        }
+
```

Comment:
if you follow the above changes I recommended, make sure to use "base" and "army". personally I'd also prefer explicitly stating else if (unit.getType().equals("army"), so it's easier to see exactly what this is doing